star-navigation/src/components/star-view/star-view.tsx

```diff
@@ -27,6 +27,20 @@ const ShutterbugSupport = () => {
   return null;
 };
 
+// This needs to be a separate component, as useThree depends on context provided by <Canvas> component.
+const TakeSnapshot: React.FC<{ onSnapshotReady: (snapshot: string ) => void }> = ({ onSnapshotReady }) => {
+  const { gl, scene, camera } = useThree();
+  useEffect(() => {
+    gl.render(scene, camera);
+    // Note that this image will be saved in the app state that is later saved in Firestore. There's 1MB limit for
+    // Firestore documents, so image size is important (we'll have four of them). JPEG with 0.25 ensures reasonable
+    // quality and the size is around 35-50kB, so very safe. In comparison, PNG is around 500kB, so it's not an option.
+    const image = gl.domElement.toDataURL("image/jpeg", 0.25);
```

Comment:
I'm curious if you tried this in Chrome, Firefox and Safari to see what, if any, are the size differences in the jpeg file.

Reply:
@dougmartin, good point. I've just tested it in Chrome, Safari, and Firefox. 
They all respect the quality param (I think the most likely point of failure) and produce JPEG images with slightly different sizes but always below 50kB. So it's pretty consistent.
Even when I set the full JPEG quality (assuming that some browser could ignore it), the image was still around 150kB, so it's fine.